runtime/src/bank.rs

```diff
@@ -3641,6 +3646,21 @@ impl Bank {
             .and_then(|instruction_recorder| Rc::try_unwrap(instruction_recorder).ok())
             .map(|instruction_recorder| instruction_recorder.into_inner().deconstruct());
 
+        let post_account_state_info =
+            self.get_transaction_account_state_info(&transaction_context, tx.message());
+
+        if self
+            .feature_set
+            .is_active(&feature_set::require_rent_exempt_accounts::id())
+        {
+            Self::verify_transaction_account_state_changes(
+                &mut process_result,
+                &pre_account_state_info,
+                &post_account_state_info,
+                &transaction_context,
+            );
+        }
```

Comment:
nit: since this only does anything if `process_result.is_ok()`, it could be moved inside `process_result.map(` below

Reply:
Oh, great point. I used to check process_result for InstructionError as well, but realized that was a waste of cycles. Absolutely moveable now.